src/cryptography/x509/name.py

```diff
@@ -253,6 +304,45 @@ def rfc4514_string(self) -> str:
             attr.rfc4514_string() for attr in reversed(self._attributes)
         )
 
+    @classmethod
+    def from_rfc4514_string(cls, value: str) -> "Name":
+        """
+        Parse a RFC4514 formatted Distinguished Name string.
+        """
+        # value must start with a attributeType, which is either a descr or a
+        # numericoid. A descr is defined in RFC4512 as starting with a
+        # alphabetic char (case insensitive), numeric OIDs start with a digit.
+        if not re.match("[A-Za-z0-9]", value):
+            raise ValueError("Value does not conform to RFC 4514")
+
+        # Split the value by commas into a list of names
+        lex = shlex.shlex(value, posix=True)
```

Comment:
I'm nervous about using `shlex` here. For obvious reasons this isn't what it's designed for, so I fear there's likely to be behavioral differences in places.

Reply:
Well, there's quite a few test cases added in this PR. `shlex` docs btw state:

> This will often be useful for writing minilanguages, (for example, in run control files for Python applications) or for parsing quoted strings.

So that seems fair enough. A more sophisticated parser (duplicating code from shlex) is certainly doable, but a lot more code.